### DISCUSSION TITLE
return_citations, more details

### DISCUSSION THREAD

**Discussion by damianoneill** [perplexityai/api-discussion#32] [Ideas]
What return_citations=true, it would be useful for the citations to include a title and description for each URL as they do today in the Graphical User Interface.

It would also be useful to be able to control the number of citations that get returned.  In some cases, 5 citations is not enough.

**Comment by shubhang98** [comment]
Thanks for the feedback. We will look into it. Can you share some examples of queries where you think you need more than 5 citations?

**Reply by ssegovba** [reply]
I am using Perplexity's API to retrieve different sources for a literature review. This is a good example where the more citations I could get, the better, but currently I am limited to 5 per query. A workaround to this could be to ask the same prompt several times, but it would be ideal if users could pass an argument in the body of the request where they explicitly state the number of expected citations.

**Comment by jess-airloc** [comment]
It seems as though I am getting only three citations back from the chat completions API. I am unable to find a value control for the number of citations requested or limits to citation counts. I ran two similar queries about the same test company and they both came back with different citations, but both were limited to a count of 3. I find this odd when there are so many more included in the GUI.

I understand that the original reason for this post is now moot as the parameter for return_citations no longer does anything, but it was the most relevant post for my question.

**Comment by Mathias-RS** [comment]
+1, I would also like to have the same amount of returned citations (like in the perplexity app)
